app/src/pages/playground/ModelComboBox.tsx

```diff
@@ -1,92 +1,131 @@
-      ) {
-        models(input: { providerKey: $providerKey }) {
-          name
-        }
-      }
-    `,
-    query
-  );
-
-  const [modelInput, setModelInput] = useState("");
-
-  const items = useMemo(() => {
-    const items = data.models.map((model) => ({
+  modelName,
+  queryReference,
+  ...comboBoxProps
+}: ModelComboBoxProps & {
+  queryReference: PreloadedQuery<ModelComboBoxQuery>;
+}) {
+  const data = usePreloadedQuery(modelsQuery, queryReference);
+  const items = useMemo((): ModelItem[] => {
+    return data.models.map((model) => ({
       name: model.name,
       id: model.name,
     }));
+  }, [data.models]);
 
-    if (
-      modelInput !== "" &&
-      !items.some(
-        (model) =>
-          model.name.toLocaleLowerCase() === modelInput.toLocaleLowerCase()
-      )
-    ) {
-      items.push({ name: modelInput, id: modelInput });
-    }
-    if (
-      props.modelName &&
-      !items.some((model) => model.name === props.modelName)
-    ) {
-      items.push({ name: props.modelName, id: props.modelName });
+  const [fieldState, setFieldState] = useState({
+    selectedKey: modelName ?? null,
+    inputValue: modelName ?? "",
+  });
+
+  const onSelectionChange = (key: Key | null) => {
+    if (typeof key === "string") {
+      const item = items.find((item) => item.id === key);
+      item?.name != null && onChange(item.name);
     }
-    return items;
-  }, [data.models, modelInput, props.modelName]);
+  };
+
+  const onInputChange = (value: string) => {
+    setFieldState((prevState) => ({
+      inputValue: value,
+      selectedKey: prevState.selectedKey,
+    }));
+  };
+
+  useEffect(() => {
+    setFieldState({
+      selectedKey: modelName ?? null,
+      inputValue: modelName ?? "",
+    });
+  }, [modelName]);
+
   return (
     <ComboBox
-      label={"Model"}
       size="L"
       data-testid="model-picker"
-      // Fallback to empty string here otherwise the picker will complain about switching from a controlled to uncontrolled component
-      // It can't distinguish between undefined and intentionally null
-      selectedKey={props.modelName ?? ""}
+      selectedKey={fieldState.selectedKey}
       aria-label="model picker"
-      onInputChange={setModelInput}
-      onSelectionChange={(key) => {
-        if (typeof key === "string") {
-          onChange(key);
+      onInputChange={onInputChange}
+      inputValue={fieldState.inputValue}
+      onSelectionChange={onSelectionChange}
+      width={"100%"}
+      allowsCustomValue
+      onBlur={() => {
+        if (fieldState.inputValue !== "") {
+          onChange(fieldState.inputValue);
+        }
+      }}
+      onKeyUp={(e) => {
+        if (e.key === "Enter") {
+          onChange(fieldState.inputValue);
         }
       }}
-      width={"100%"}
-      {...props}
-      defaultItems={items}
       menuTrigger="focus"
       container={container}
+      description={"Select a model or type one in and hit enter"}
```

Comment:
```suggestion
      description={"Choose a model name from the list, or enter a custom model name"}
```

What do you think of this? I think the Enter key / Blur is intuitive enough to leave off of the instructions

Reply:
Yup i'm down, i added teh hit enter but didn't feel great about it, if you think it's intuitive enough will leave it off